polygeist/tools/cgeist/Lib/CGExpr.cc

```diff
@@ -1189,7 +1189,16 @@ ValueCategory MLIRScanner::VisitDeclRefExpr(DeclRefExpr *E) {
                            /*isReference*/ true);
     }
 
-    auto gv = Glob.GetOrCreateGlobal(VD, /*prefix=*/"");
+    // We need to decide where to put the Global.  If we are in a device
+    // function, the global should be in the gpu module (which is nested inside
+    // another main module).
+    std::pair<mlir::memref::GlobalOp, bool> gv;
+    if (function->getParentOp() != module.get()) {
+      // Device function
+      gv = Glob.GetOrCreateGlobal(VD, /*prefix=*/"", true, true);
```

Comment:
Do you think it would be clearly to pass `FunctionContext` instead of a bool?

Reply:
Thanks for the suggestion.  Can you please elaborate why you think passing `FunctionContext` will be better here?

Reply:
`FunctionContext` is an enum with two possible values, so it is in some sense very similar to a boolean, but it is explicit on what it means.
In here, you pass true, but it is not clear what it means without looking at how it is used.

Reply:
Hmm, if I am passing `FunctionContext` here with an intended behavior like a Boolean, how would that help clarify (without looking at how it is used)?

Reply:
Do you think `gv = Glob.GetOrCreateGlobal(VD, /*prefix=*/"", true, FunctionContext::SYCLDevice);` tells more than `gv = Glob.GetOrCreateGlobal(VD, /*prefix=*/"", true, true);`? 
It is clearer now that it is no longer passing `true` or `false`.

Reply:
OK, that makes sense, thank you